source/common/http/http2/codec_impl.cc

```diff
@@ -539,18 +544,24 @@ int ConnectionImpl::onInvalidFrame(int32_t stream_id, int error_code) {
   ENVOY_CONN_LOG(debug, "invalid frame: {} on stream {}", connection_, nghttp2_strerror(error_code),
                  stream_id);
 
+  StreamImpl* stream = getStream(stream_id);
   // The stream is about to be closed due to an invalid header or messaging. Don't kill the
   // entire connection if one stream has bad headers or messaging.
   if (error_code == NGHTTP2_ERR_HTTP_HEADER || error_code == NGHTTP2_ERR_HTTP_MESSAGING) {
     stats_.rx_messaging_error_.inc();
-    StreamImpl* stream = getStream(stream_id);
     if (stream != nullptr) {
       // See comment below in onStreamClose() for why we do this.
       stream->reset_due_to_messaging_error_ = true;
     }
     return 0;
   }
 
+  if (stream != nullptr) {
```

Comment:
Thanks for adding the new test! 

I'm sold on the test being useful but for the codec changes, I'd prefer rather than the codec have to know the behavior of the HCM, that the HCM actually trigger this by doing something like onDecoderDestroy to the codec if we think the lifetime of the streams in HCM and codec differs.  

Alternate if the lifetime of the streams does not differ, I think this PR adds more risk than it removes and we should simply add the test.

Reply:
Thanks for the review! I see. Reverted the codec change.